File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    answer = 0
    lst = deque()
    for idx in range(len(priorities)):
        lst.append([idx, priorities[idx]])
    while lst:
        maxP = max(lst, key = lambda x : x[1])[1]
        job = lst.popleft()
        if maxP == job[1]: # 현재 작업이 가장 높은 우선 순위를 가진 경우
            answer += 1
            if job[0] == location:  # 원하는 작업이 완료된 경우
                break
        else:
            lst.append(job)
    return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    var answer = 0;
    let queue = [];
    for (let i = 0; i < priorities.length; i ++) {
        queue.push([i, priorities[i]])
    }
    let temp = [];
    while (queue.length > 0) {
        measureMax = [...queue]
        measureMax.sort((a, b) => b[1] - a[1]); // 우선 순위가 높은 것부터 정렬
        let maxVal = measureMax[0][1]
        if (maxVal === queue[0][1]) {
            answer += 1
            if (queue[0][0] === location) {
                break
            }
            queue.shift()
        } else {
        queue.push((queue.shift()))
    }
        
    } 
    console.log(queue)
    return answer;
}